block_cascade/cli/crud.py

```diff
@@ -0,0 +1,259 @@
+    }
+
+    if CPU_REPLICA_COUNT > 0 and GPU_REPLICA_COUNT > 0:
+        RESOURCE_POOLS = [CPU_POOL, GPU_POOL]
+    elif CPU_REPLICA_COUNT > 0:
+        RESOURCE_POOLS = [CPU_POOL]
+    elif GPU_REPLICA_COUNT > 0:
+        RESOURCE_POOLS = [GPU_POOL]
+    else:
+        RESOURCE_POOLS = []
+
+    PERSISTENT_RESOURCE = {
+        "display_name": resource.persistent_resource_id,
+        "resource_pools": RESOURCE_POOLS,
+    }
+
+    return PERSISTENT_RESOURCE
+
+
+def get_parent_str(resource: GcpResource) -> str:
+    """Return the parent string for the given project and region."""
+    project_id = resource.environment.project
+    region = resource.environment.region
+    return f"projects/{project_id}/locations/{region}"
+
+
+def get_endpoint_str(region: str) -> str:
+    """Return the parent string for the given project and region."""
+    return f"{region}-{SERVICE}"
+
+
+@click.command()
+@click.option(
+    "--config-name",
+    "-c",
+    required=True,
+    help="Name of the configuration block in cascade.yml to use.",
+)
+def create_persistent_resource(config_name: str):
+    """
+    Create a persistent resource from a configuration block in cascade.yaml.
+    Configuration block will be keyed on the config-name argument.
+    """
+    resource = find_default_configuration()[config_name]
+    persistent_resource = get_persistent_resource_payload(resource)
+
+    # CreatePersistentResoure SDK
+    region = resource.environment.region
+    client_options = {"api_endpoint": f"{region}-{SERVICE}"}
+    client = aiplatform.PersistentResourceServiceClient(client_options=client_options)
+
+    parent = get_parent_str(resource)
+
+    try:
+        operation = client.create_persistent_resource(
+            parent=parent,
+            persistent_resource_id=resource.persistent_resource_id,
+            persistent_resource=persistent_resource,
+        )
+        click.echo(
+            """
+            Attempting to create Persistent Resource... 
+            This may take several minutes and will continue if you close this terminal.
+            """
+        )
+        response = operation.result()
+        response = json.loads(json_format.MessageToJson(response._pb))
+        click.echo(f'Persistent resource {response["name"]} created successfully.')
+    except Exception as e:
+        click.echo(e)
+
+
+@click.command()
+@click.option(
+    "--project",
+    "-p",
+    help="GCP project ID. Inferred from gcloud config if not provided.",
+)
+@click.option(
+    "--region",
+    "-r",
+    help="GCP region. Inferred from gcloud config if not provided.",
+)
+def list_persistent_resources(region: str = None, project: str = None):
+    # try to infer project and region from gcloud config
+    if region is None or project is None:
+        gcp_config = get_gcloud_config()
+        if project is None:
+            project = get_gcp_project(gcp_config)
+        if region is None:
+            region = get_gcp_region(gcp_config)
+
```

Comment:
Should exit early if `project` and/or `region` cannot be read from `gcloud` configuration.